course_info/icommons.py

```diff
@@ -186,9 +186,10 @@ def _get_course_instance_id(self, course_instances):
                     return None
                 _, id_ = self._parse_type_and_id_from_url(
                              ci['primary_xlist_instances'][0])
-                primary_cids.add(id_)
+                primary_cids.add(id_)  # unicode string
             else:
-                primary_cids.add(ci['course_instance_id'])
+                # must also be unicode string to match any dupes found above
+                primary_cids.add(u'{}'.format(ci['course_instance_id']))
```

Comment:
That does sound better! :) If you don't mind, we'll let the existing code go out, and I'll file this in TIL, should do this next time.